### ISSUE TITLE
Inheriting a Razor component from a generic class with a nullable type parameter results in CS8669

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#85384]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I feel like this is dotnet/aspnetcore#32733 but this has been fixed some time ago.

When I derive a Razor component from a generic base and use a nullable reference type as the type parameter, there is a CS8669 warning (or error when warnings are treated as errors)

### Expected Behavior

CS8669 should not be triggered

### Steps To Reproduce

TestBase.cs:
```cs
public class TestBase<T> : ComponentBase
{
    public T Prop { get; set; } = default!;
}
```

Test.razor
```razor
@inherits TestBase<string?>

<div>@Prop</div>
```

### Exceptions (if any)

`…\Microsoft.NET.Sdk.Razor.SourceGenerators\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\Test_razor.g.cs(572,48,572,49): error CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source.`

### .NET Version

6.0.300-preview.22204.3 on 6.0.4

### Anything else?

VS 17.2.0 Preview 3.0
